File: backend/app/schemas/conversation.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field

from app.models.message import SenderType


class MessageCreate(BaseModel):
    body: str = Field(min_length=1, max_length=2000)


class MessageRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    sender_type: SenderType
    body: str
    created_at: datetime
    read_at: datetime | None


class ConversationClientSummary(BaseModel):
    id: uuid.UUID
    full_name: str
    avatar_url: str | None


class ConversationRead(BaseModel):
    id: uuid.UUID
    client: ConversationClientSummary
    last_message_at: datetime | None
    created_at: datetime


class ConversationBarbershopSummary(BaseModel):
    id: uuid.UUID
    name: str
    logo_url: str | None


class ClientConversationCreate(BaseModel):
    tenant_id: uuid.UUID


class ClientConversationRead(BaseModel):
    id: uuid.UUID
    barbershop: ConversationBarbershopSummary
    last_message_at: datetime | None
    created_at: datetime

File: frontend/src/types/conversation.ts
export interface ConversationClientSummary {
  id: string;
  full_name: string;
  avatar_url: string | null;
}

export interface Conversation {
  id: string;
  client: ConversationClientSummary;
  last_message_at: string | null;
  created_at: string;
}

export interface Message {
  id: string;
  sender_type: "CLIENT" | "STAFF";
  body: string;
  created_at: string;
  read_at: string | null;
}
